dencam/sunsaver_log.py

```diff
@@ -0,0 +1,77 @@
+import minimalmodbus
+import math
+import csv
+import os
+from serial import SerialException
+from datetime import datetime
+
+
+def get_free_space(media_path):
+    try:
+        statvfs = os.statvfs(media_path)
+        bytes_available = statvfs.f_frsize * statvfs.f_bavail
+        gigabytes_available = bytes_available/1000000000
+        return gigabytes_available
+    except FileNotFoundError:
+        return 0
+
+
+def log_solar_info():
+    storage_limit = 0.5
+    solar_list = ['init']
+    now = datetime.now()
+    date_time_string = now.strftime("%Y-%m-%d_%Hh%Mm%Ss")
+    try:
+        SunSaver = minimalmodbus.Instrument('/dev/ttyUSB0', 1)
+    except SerialException:
+        solar_list = [date_time_string + ' USB PORT ERROR']
+    SunSaver.serial.baudrate = 9600
+    SunSaver.serial.stopbits = 2
+    try:
+        volt_batt = str(math.trunc(SunSaver.read_register(8) * 100 * 2**-(15)))
+        volt_arr = str(SunSaver.read_register(9) * 100 * 2**-(15))
+        volt_ld = str(math.trunc(SunSaver.read_register(10) * 100 * 2**-(15)))
+        curr_chrg = str(math.trunc(SunSaver.read_register(11) * 79.16 * 2**-(15)))
+        curr_ld = str(math.trunc(SunSaver.read_register(12) * 79.16 * 2**-(15)))
+        temp_amb = str(SunSaver.read_register(15))
+        temp_rts = str(SunSaver.read_register(16))
+        charge_state = str(SunSaver.read_register(17))
+        ah_charge = str(SunSaver.read_register(45) * 0.1)
+        ah_load = str(SunSaver.read_register(46) * 0.1)
+        alarm = SunSaver.read_register(50)
+        alarm_list = ["RTS open", "RTS shorted", "RTS disconnected",
```

Comment:
variables like these lists can be declared outside of the function as global variables so the pi doesnt have to create them every time the function is called